### ISSUE TITLE
Error in IonicCordova.deploy.deleteVersion in iOS

### ISSUE THREAD

**Issue by deliverymanager** [ionic-team/cordova-plugin-ionic#38]
When I try to delete a snapshot with the API that already exists inside the app in iOS, then I always get the following error.

```
2018-01-04 14:50:00.237956+0200  deleting snapshots 923d5a02-bc9c-4bca-abe5-5509d067ad2e
2018-01-04 14:50:00.247889+0200  *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArrayM objectAtIndex:]: index 1 beyond bounds [0 .. 0]'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010c0c912b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010b72ff41 objc_exception_throw + 48
	2   CoreFoundation                      0x000000010c1090cc _CFThrowFormattedException + 194
	3   CoreFoundation                      0x000000010bff8eaf -[__NSArrayM objectAtIndex:] + 175
	4   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x00000001073aaaac -[IonicDeploy deleteVersion:] + 92
	5   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x000000010744fa66 -[CDVCommandQueue execute:] + 870
	6   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x000000010744f2d1 -[CDVCommandQueue executePending] + 833
	7   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x000000010744594a __40-[CDVCommandDelegateImpl evalJsHelper2:]_block_invoke + 314
	8   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x000000010745233c -[CDVUIWebViewEngine evaluateJavaScript:completionHandler:] + 156
	9   Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x00000001074457c4 -[CDVCommandDelegateImpl evalJsHelper2:] + 212
	10  Foundation                          0x000000010b16e1e8 __NSThreadPerformPerform + 334
	11  CoreFoundation                      0x000000010c06c101 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	12  CoreFoundation                      0x000000010c10bf71 __CFRunLoopDoSource0 + 81
	13  CoreFoundation                      0x000000010c050a19 __CFRunLoopDoSources0 + 185
	14  CoreFoundation                      0x000000010c04ffff __CFRunLoopRun + 1279
	15  CoreFoundation                      0x000000010c04f889 CFRunLoopRunSpecific + 409
	16  GraphicsServices                    0x000000010f3699c6 GSEventRunModal + 62
	17  UIKit                               0x0000000107afe5d6 UIApplicationMain + 159
	18  Œ§Œø œáŒ±ÃÅŒΩŒπ œÑŒøœÖ Œ£œÑŒ±œÖÃÅœÅŒøœÖ 0x000000010739ec41 main + 65
	19  libdyld.dylib                       0x000000010d3c8d81 start + 1
	20  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

ionic info
cli packages: (/usr/local/lib/node_modules)

@ionic/cli-utils  : 1.19.0
ionic (Ionic CLI) : 3.19.0
global packages:

cordova (Cordova CLI) : 8.0.0 
local packages:

Cordova Platforms : ios 4.5.4
Ionic Framework   : ionic1 1.3.3
System:

Android SDK Tools : 26.1.1
Node              : v6.11.4
npm               : 3.10.10 
OS                : macOS High Sierra
Xcode             : Xcode 9.2 Build version 9C40b 
Environment Variables:

ANDROID_HOME : /Users/adomvris/Library/Android/sdk/
Misc:

backend : pro

**Comment by Maistho** [comment]
This line is responsible:

https://github.com/ionic-team/cordova-plugin-ionic/blob/master/src/ios/IonicDeploy.m#L724

Should be `objectAtIndex:0`?

@Fuiste probably knows what's up.

**Comment by ztecharoberts** [comment]
Is this fixed in the latest update? I currently have a large number of app crashes due to this defect.

**Comment by ztecharoberts** [comment]
Submitted a pull request as this is a hard crash for me in prod and is a massive issue.

**Comment by Fuiste** [comment]
Thanks for the PR! Merged and launched in 4.0.1
